Tests/SonarScanner.MSBuild.PreProcessor.Test/CacheProcessorTests.cs

```diff
@@ -327,5 +338,21 @@ public CacheContext(CacheProcessorTests owner, string commandLineArgs = "/k:key"
             public void Dispose() =>
                 Sut.Dispose();
         }
+
+        private sealed class FileData
+        {
+            public string RelativeFilePath { get; }
```

Comment:
After addressing the other comments (remove encoding, make it a local variable), this will be holder for 2 values. Record would do a better job as you said.

Can we (easily) use just a list of tuples instead?

Reply:
Yes, I used named tuples and the code is still readable.